File: BinaryTrees/15/maxDepth.js
class TreeNode {
  constructor(val = 0, left = null, right = null) {
    this.val = val;
    this.left = left;
    this.right = right;
  }
}

const maxDepth = (root) => {
  if (root) {
    return Math.max(1 + maxDepth(root.left), 1 + maxDepth(root.right));
  }
  return 0;
};

File: BinaryTrees/15/maxDepth.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def maxDepth(self, root: TreeNode) -> int:
        if(root):
            return max(1 + self.maxDepth(root.left), 1 + self.maxDepth(root.right))
        return 0
